sphinx/themes/basic/static/searchtools.js

```diff
@@ -466,14 +466,18 @@ const Search = {
       // add support for partial matches
       if (word.length > 2) {
         const escapedWord = _escapeRegExp(word);
-        Object.keys(terms).forEach((term) => {
-          if (term.match(escapedWord) && !terms[word])
```

Comment:
`terms[word]` will be falsey (`0`) for the first document.

Reply:
@wlach based on index-format findings (and test coverage), should this be rephrased to:

> `terms[word]` will be falsey (`0`) for a document with ID zero that exists in no other documents

(kinda wordy, but captures the fact that `[0]` is _not_ falsy)

Reply:
(edited with a correction and for conciseness)